pic-sure-util/src/main/java/edu/harvard/dbmi/avillach/util/HttpClientUtil.java

```diff
@@ -275,22 +277,24 @@ public static HttpResponse simpleGet(HttpClient client, String uri, Header... he
 	}
 
 	public static InputStream simpleGet(String uri, HttpClient client, Header... headers) {
+		return simpleGetWithConfig(uri, client, null, headers);
+	}
+
+	public static InputStream simpleGetWithConfig(
+		String uri, HttpClient client, RequestConfig config, Header... headers
+	) throws ApplicationException {
 		HttpGet get = new HttpGet(uri);
 		get.setHeaders(headers);
-
-		HttpResponse response;
-
-		try {
-			response = client.execute(get, buildHttpClientContext());
-		} catch (IOException ex) {
-			logger.error("InputStream simpleGet() cannot get response by GET from url: {} - " + ex.getLocalizedMessage(), uri);
-			throw new ApplicationException("Inner problem, please contact system admin and check the server log");
+		if (config != null) {
```

Comment:
We had consecutive try-catches that caught the same exception type, and logged and handled it in the exact same manner. Combined them.